src/sentry/integrations/issues.py

```diff
@@ -143,3 +146,87 @@ def sync_status_outbound(self, external_issue, is_resolved, project_id, **kwargs
         Propagate a sentry issue's status to a linked issue's status.
         """
         raise NotImplementedError
+
+    def should_resolve(self, configuration, data):
```

Comment:
The `should_resolve` and `should_unresolve` methods have me a bit confused. Could you give me an example of how these would be used? Is there a reason they are separate methods? If so, what would be the differences between the two?